docs/kratos/emails-sms/05_custom-email-templates.mdx

```diff
@@ -0,0 +1,285 @@
+- `recovery.invalid` - `To` for invalidating recovery.
+- `verification.valid` - `To`, `VerificationURL` and `Identity` for validating verification.
+- `verification.invalid` - `To` for invalidating verification.
+
+For example:
+
+[`https://github.com/ory/kratos/blob/master/courier/template/courier/builtin/templates/verification/valid/email.body.gotmpl`](https://github.com/ory/kratos/blob/master/courier/template/courier/builtin/templates/verification/valid/email.body.gotmpl)
+
+```gotmpl title="courier/template/templates/verification/valid/email.body.gotmpl"
+Hi, please verify your account by clicking the following link:
+
+<a href="{{ .VerificationURL }}">{{ .VerificationURL }}</a>
+```
+
+```gotmp title="courier/template/templates/verification/valid/email.body.plaintext.gotmpl"
+Hi, please verify your account by clicking the following link: {{ .VerificationURL }}
+```
+
+## Customizing template content for specific users
+
+To enable customizing the content of templates based on the identity of the recipient of the email, the Ory identity is available
+as the `Identity` object. This object is a map containing all the attributes of an identity, such as `id`, `state`,
+`recovery_addresses`, `verifiable_addresses` and `traits`.
+
+### Nested templates: i18n customization
+
+You can use nested templates to render `email.subject.gotmpl`, `email.body.gotmpl` and `email.body.plaintext.gotmpl` templates.
+
+````mdx-code-block
+<Tabs>
+  <TabItem value="cloud" label="Ory Cloud" default>
+
+You can either use in-line template definitions in nested templates. The following example defines the message body for verification emails.
+It assumes that there is a `lang` attribute defined in the identity traits that contains the preferred language of the user. The verification email
+content changes if the system detects `fr` or `it` in the identity traits. If the system detects any other language, it uses the `eng` template.
+
+```gotmpl
+{{define "en"}}
+Hi,
+
+Please recover access to your account by clicking the following link:
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{define "fr"}}
+Bonjour ,
+
+Veuillez récupérer l'accès à votre compte en cliquant sur le lien suivant :
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{define "it"}}
+Ciao ,
+
+Recuperare l'accesso al proprio account facendo clic sul seguente link:
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{- else if eq .Identity.traits.language "fr" -}}
+{{ template "fr" . }}
+{{- else if eq .Identity.traits.language "it" -}}
+{{ template "it" . }}
+{{- else -}}
+{{ template "en" . }}
+{{- end -}}
+```
+
+  </TabItem>
+  <TabItem value="selfhosted" label="Self-Hosted Ory Kratos Config" default>
+
+To create nested templates, you can use either in-line template definitions, or as in this example, use separate templates. The following
+example defines the message body for verification emails. It assumes that there is a `lang` attribute defined in the identity traits
+that contains the preferred language of the user.
+
```

Comment:
I think also add a warning here about using separate templates in Ory Cloud. They aren't supported since you need to specify the `template_override_path` and this cannot be specified in Cloud.

Reply:
Same comment as in the `template_override_path` case. Additionally, `template_override_path` is not in the Cloud config by default, you'd have to go out of your way to add it manually.

Reply:
Well people got confused in our Slack channel on what is possible in Cloud due to what is possible in Open Source. I think just strengthening what you can and cannot do is always good since it leads to less repetitive questions in slack and GH. 

As I mentioned in a previous comment, maybe adding what isn't supported in Cloud at the top of this section might then be best, since you aren't repeating yourself across all of the docs and it is in one place for these particular differences.